File: src/test/java/com/twsg/lnl/kotlin/c5_delegate/observable/g1_java/BankAccountTest.java
package com.twsg.lnl.kotlin.c5_delegate.observable.g1_java;

import org.junit.Test;

import java.util.function.BiConsumer;

import static org.mockito.Mockito.*;

/**
 * @author tonyl
 */
public class BankAccountTest {

    @Test
    public void setBalance_should_notify_observer() throws Exception {
        BiConsumer<Double, Double> observer = mock(BiConsumer.class);

        BankAccount account = new BankAccount(0, observer);
        account.setBalance(5D);

        verify(observer, atLeast(1)).accept(0D, 5D);
    }

}

File: src/test/java/com/twsg/lnl/kotlin/c5_delegate/observable/g2_kotlin/BankAccountTest.kt
package com.twsg.lnl.kotlin.c5_delegate.observable.g2_kotlin

import org.junit.Test
import org.mockito.Mockito.*

/**
 * @author tonyl
 */
class BankAccountTest {

    @Test
    fun setBalance_should_notify_observer() {
        abstract class Func : ((Double, Double) -> Unit)

        val observer: (Double, Double) -> Unit = mock(Func::class.java)

        val account = BankAccount(0.0, observer)
        account.balance = 5.0

        verify(observer, atLeast(1)).invoke(0.0, 5.0)
    }

}